iop/fs/bdm/src/part_driver.c

```diff
@@ -83,8 +87,25 @@ static int part_getPartitionTable(struct block_device* bd, part_table* part)
     if (sbuf[0x1FE] == 0x55 && sbuf[0x1FF] == 0xAA) {
         for (i = 0; i < 4; i++) {
             part_raw = (part_raw_record*)(sbuf + 0x01BE + (i * 16));
-            if (part_getPartitionRecord(bd, part_raw, &part->record[i]) != 0)
-                return 0; //Invalid record encountered, so the table is probably invalid.
+            if (part_getPartitionRecord(bd, part_raw, &part->record[i]) == 0) {
+                if (i == 0) {
+                    /* even 1st record fail - that means we have no MBR, but maybe we have removable FAT
+                     "Removable" has no partition table, so we will create 1 virtual partition holding whole usb.
+                     back to the hardcoded fat32 values
+                     sid = 0x0c, start - 0x00, sectorCount - ????
+                     TODO: is there a way to calculate USB size for completely empty drives?
+                     1 partition
+                     TODO: do we really need to care about sid and sectorCount?
+                     FAT driver will ignore those values anyway */
+                    part->record[i].sid = 0x0C;
+                    part->record[i].start = 0;
+                    // part->record[i].count = sbuf[0x20] + (sbuf[0x21] << 8) + (sbuf[0x22] << 16) + (sbuf[0x23] << 24);
+                    part->record[i].count = 0xFFFFFFFF; // maybe just assign hardcoded maximum value; FAT driver will use its own probing anyway
+                }
+                // if we have magic 0x55 0xAA, partition records cannot fail, we will have at least one removable storage
+                // that will be expanded into mass0p0 for example
+                return (i + 1);
+            }
```

Comment:
If there is no partition table, then the entire block device can be mounted to the FAT driver directly, without an intermediate partition driver.

If there is a partition table then:
- usb0p0 = created by usbmass_bd driver, mounted to MBR file system
- usb0p1 = created by MBR file system, mounted to FAT32 file system

If there is NO partition table then:
- usb0p0 = created by usbmass_bd driver, mounted to FAT32 file system

This change should be removed.

Reply:
Ah thanks I just didnt found that if there is no partition table - then it will fallback to Removable device. If it is true, then this is not needed.

Reply:
yes, I understand that - I just didn't found that switch. I thought that it is doing things in such a way:
check if usb0p0 is MBR
fail